src/bodywork/k8s/service_deployments.py

```diff
@@ -402,16 +404,15 @@ def list_service_stage_deployments(namespace: str) -> Dict[str, Dict[str, str]]:
             "git_url": deployment.spec.template.spec.containers[0].args[0],
             "git_branch": deployment.spec.template.spec.containers[0].args[1],
             "has_ingress": (
-                "true" if has_ingress(namespace, deployment.metadata.name) else "false"
+                has_ingress(deployment.metadata.namespace, deployment.metadata.name)
```

Comment:
This won't return a string, it'll return a bool - think we should leave this one as-is.

Reply:
I'm amazed MyPy didn't flag this, because the return type is `Dict[str, str]`.

Reply:
That's because a bool will automatically convert to a string because the bool type has a "to_string()" function (or whatever the Python equivalent of that is) therefore this will return a string with either "True" or "False".

Reply:
No it won't - there's no such thing in Python. At most, `""` equates to `False` and `"anystring"` equates to `True`, but that's it.

Reply:
There seems to be some native translation of bools to strings because this works. Try typing the following into a Python Console and look at the result:

str(bool(True))
str(bool(False))

It will return strings with `True` and `False` respectively. This is effectively what is happening in this code snippet. `has_ingress()` returns a bool not a string.

Reply:
But you have to force the conversion - e.g. by calling `str(obj)`. It won't happen automatically otherwise - the Python interpreter completely ignores type hints.

Reply:
Change the code from my previous comment with `print(bool(True))` or do `f"{bool(True)}"` and you will get the same result. It returns the True/False string value that we require. 

I see no value add by explicitly declaring the strings in a conditional statement and it's not as simple. It's effectively a custom implementation of casting a bool to a string.

Reply:
Got it - yes. And this'll be replaced by a DTO at some point, in which case we want the `bool` anyway.

Should probably change `Dict[str, str]` -> `Dict[str, Union[str, bool]]` or `Dict[str, Any]` to that it's accurate.

Reply:
I've changed it to `Dict[str, Any]`, we also have `int` types in there as well e.g. available_replicas.